src/overlays/actors/ovl_En_Po_Composer/z_en_po_composer.c

```diff
@@ -100,88 +128,619 @@ static DamageTable D_80BC67E8 = {
+            this->csIds[i] = csId;
+            csId = CutsceneManager_GetAdditionalCsId(csId);
+        }
+
+        if (CHECK_WEEKEVENTREG(WEEKEVENTREG_14_02)) {
+            this->sharpCsNum = SHARP_CS_ENCOUNTER_OTHER;
+        } else {
+            this->sharpCsNum = SHARP_CS_ENCOUNTER_FIRST;
+        }
+        this->inCutscene = false;
+        EnPoComposer_SetupAwaitPlayer(this);
+    }
+}
+
+void EnPoComposer_Destroy(Actor* thisx, PlayState* play) {
+    EnPoComposer* this = THIS;
+
+    LightContext_RemoveLight(play, &play->lightCtx, this->lightNode);
+    Collider_DestroyJntSph(play, &this->lanternCollider);
+    Collider_DestroyCylinder(play, &this->bodyCollider);
+}
+
+void EnPoComposer_SetupAwaitPlayer(EnPoComposer* this) {
+    Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, PO_COMPOSER_ANIM_IDLE_STILL);
+    this->actionFunc = EnPoComposer_AwaitPlayer;
+}
+
+void EnPoComposer_AwaitPlayer(EnPoComposer* this, PlayState* play) {
+    if (this->actor.xzDistToPlayer < 200.0f) {
+        EnPoComposer_SetupStartCutscene(this);
+    }
+}
+
+void EnPoComposer_SetupStartCutscene(EnPoComposer* this) {
+    this->inCutscene = true;
+    this->actionFunc = EnPoComposer_StartCutscene;
+}
+
+void EnPoComposer_StartCutscene(EnPoComposer* this, PlayState* play) {
+    if (CutsceneManager_IsNext(this->csIds[this->sharpCsNum])) {
+        CutsceneManager_Start(this->csIds[this->sharpCsNum], &this->actor);
+        EnPoComposer_SetupStartedCutscene(this);
+    } else {
+        if (CutsceneManager_GetCurrentCsId() == CS_ID_GLOBAL_TALK) {
+            CutsceneManager_Stop(CS_ID_GLOBAL_TALK);
+        }
+        CutsceneManager_Queue(this->csIds[this->sharpCsNum]);
+    }
+}
+
+void EnPoComposer_SetupStartedCutscene(EnPoComposer* this) {
+    this->actionFunc = EnPoComposer_StartedCutscene;
+}
+
+void EnPoComposer_StartedCutscene(EnPoComposer* this, PlayState* play) {
+}
+
+void EnPoComposer_SetupPlayCurse(EnPoComposer* this) {
+    this->actionTimer = 0;
+    this->actor.flags |= ACTOR_FLAG_TARGETABLE;
+    Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimationInfo, PO_COMPOSER_ANIM_PLAYING_CURSE);
+    this->actionFunc = EnPoComposer_PlayCurse;
+}
+
+void EnPoComposer_PlayCurse(EnPoComposer* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+
+    // Damage the player every second
+    if (this->actionTimer == 20) {
+        func_800B8D10(play, &this->actor, 0.0f, 0, 0.0f, 1, 4);
+        this->actionTimer = 0;
+    } else {
+        this->actionTimer++;
+    }
+
+    // Ocarina check
+    if (player->stateFlags2 & PLAYER_STATE2_8000000) {
+        if (!sPlayerIsPlayingOcarina) {
+            // Play sound whenever the player begins playing the Ocarina
+            Audio_PlaySfx(NA_SE_SY_TRE_BOX_APPEAR);
+        }
+        sPlayerIsPlayingOcarina = true;
+    } else {
+        sPlayerIsPlayingOcarina = false;
+    }
+
+    // Song check, check if player just played song of storms or song of healing to react to
```

Comment:
Extreme nitpick, but the end of this sentence reads awkwardly and frankly is still understandable without it?
```suggestion
    // Song check, check if player just played song of storms or song of healing
```